gtsam/hybrid/GaussianHybridFactorGraph.cpp

```diff
@@ -115,6 +115,17 @@ pair<GaussianMixture::shared_ptr, boost::shared_ptr<Factor>> EliminateHybrid(
     return hasNull?GaussianFactorGraph():gfg;
   };
 
+  // TODO(fan): Now let's assume that all continuous will be eliminated first!
+  // Here sum is null if remaining are all discrete
+  std::cout << "[EL] " << "key = "; ordering.print("");
+  if (sum.empty()) {
+    std::cerr << "LOLOLOL you got empty! let's do discrete!\n";
+    // Not sure if this is the correct thing, but anyway!
```

Comment:
1) should do maxProduct, check that PR and code in DiscreteFactorGraph
2) You need a unit test.

Reply:
I did maxProduct, hope this is the correct way of doing it?

I also added unit test to check against partial elimination.